src/Microsoft.TemplateEngine.Abstractions/Installer/IInstaller.cs

```diff
@@ -0,0 +1,84 @@
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.TemplateEngine.Abstractions.GlobalSettings;
+using Microsoft.TemplateEngine.Abstractions.TemplatePackages;
+
+namespace Microsoft.TemplateEngine.Abstractions.Installer
+{
+    public interface IInstaller
+    {
+        /// <summary>
+        /// User can specify name of specific installer to be used to install package.
+        /// e.g: nuget, folder, vsix(to download from VS marketplace), npm, maven...
+        /// This is useful when installer can't be determined based on <see cref="InstallRequest.Identifier"/> and <see cref="InstallRequest.Details"/>
+        /// </summary>
+        Guid FactoryId { get; }
+
+        /// <summary>
+        /// User can specify name of specific installer to be used to install package.
+        /// e.g: nuget, folder, vsix(to download from VS marketplace), npm, maven...
+        /// This is useful when installer can't be determined based on <see cref="InstallRequest.Identifier"/> and <see cref="InstallRequest.Details"/>
+        /// </summary>
+        string Name { get; }
+
+        IManagedTemplatePackagesProvider Provider { get; }
+
+        /// <summary>
+        /// Installer should determine if it can install specific <see cref="InstallRequest"/>.
+        /// Ideally it should as far as calling backend server to determine if such identifier exists.
+        /// </summary>
+        Task<bool> CanInstallAsync(InstallRequest installationRequest, CancellationToken cancellationToken);
+
+        /// <summary>
+        /// Deserializes <see cref="TemplatePackageData"/> to <see cref="IManagedTemplatePackage"/>
+        /// </summary>
+        /// <param name="provider">The provider that provides the data</param>
+        /// <param name="data">Data to serialize</param>
+        /// <returns>deserialized <see cref="IManagedTemplatePackage"/></returns>
+        IManagedTemplatePackage Deserialize(IManagedTemplatePackagesProvider provider, TemplatePackageData data);
+
+        /// <summary>
+        /// Gets latest versions for <paramref name="sources"/>
+        /// </summary>
+        /// <param name="sources">sources to get latest versions for</param>
+        /// <returns>list of <see cref="CheckUpdateResult"/> containing latest versions for the sources</returns>
+        /// <param name="cancellationToken"></param>
+        Task<IReadOnlyList<CheckUpdateResult>> GetLatestVersionAsync(IEnumerable<IManagedTemplatePackage> sources, CancellationToken cancellationToken);
+
+        /// <summary>
+        /// Installs the template source
+        /// </summary>
+        /// <param name="installRequest">details for installation</param>
+        /// <returns><see cref="InstallResult"/> containing installation results and <see cref="IManagedTemplatePackage"/> if installation was successful</returns>
```

Comment:
The part with `and <see cref="IManagedTemplatePackage"/>` makes no sense for me here.

Reply:
I've changed it to:
`        /// <returns><see cref="InstallResult"/> containing installation results and installed <see cref="InstallResult.TemplatePackage"/> if installation was successful.</returns>` 
Does it make more sense now?